File: Prosperecon/marketing/conteudo/carrossel-vender-carta-2026-05-29/render.py
from playwright.sync_api import sync_playwright
import os

BASE = os.path.dirname(os.path.abspath(__file__))
HTML = f"file://{BASE}/carrossel.html"
OUT  = os.path.join(BASE, "instagram")
os.makedirs(OUT, exist_ok=True)

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page(viewport={"width": 1080, "height": 1350})
    page.goto(HTML, wait_until="networkidle")

    slides = page.query_selector_all(".slide")
    print(f"Encontrei {len(slides)} slides")

    for i, slide in enumerate(slides, 1):
        path = os.path.join(OUT, f"slide-{i:02d}.png")
        slide.screenshot(path=path)
        print(f"  ✓ slide-{i:02d}.png")

    browser.close()
    print(f"\nPronto — {len(slides)} PNGs em instagram/")

File: marketing/conteudo/carrossel-investimento-2026-06-18/render.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.setViewportSize({ width: 1080, height: 1350 });

  const htmlPath = 'file://' + path.resolve(__dirname, 'carrossel.html');
  await page.goto(htmlPath, { waitUntil: 'networkidle' });

  const slides = await page.$$('.slide');
  for (let i = 0; i < slides.length; i++) {
    const num = String(i + 1).padStart(2, '0');
    await slides[i].screenshot({
      path: path.join(__dirname, 'instagram', `slide-${num}.png`),
      type: 'png'
    });
    console.log(`slide-${num}.png`);
  }

  await browser.close();
  console.log(`\nDone — ${slides.length} slides rendered.`);
})();
